### DISCUSSION TITLE
Wrist detection

### DISCUSSION THREAD

**Discussion by jackdur** [InfiniTimeOrg/InfiniTime#2192] [Ideas]
Please add wrist detection – functionality to detect when the watch is/isn't on the wrist. This probably requires an algorithm that uses the heart rate sensor and is at the same time enough but not too sensitive to detect the loss of led return reflect.

Wrist detection is needed to automatically lock the watch with a password when you take it off.

**Comment by mark9064** [comment]
Sorry I forgot to comment on the original issue(s) - feature requests are tracked in discussion posts so I've converted both of your issues over - we've found the discussion board works much better for fleshing out ideas, hope this is ok :)

**Reply by jackdur** [reply]
Absolutely, thank you! :)

Btw what do you think, is this feature possible to implement in InfiniTime?

**Reply by mark9064** [reply]
It's hard to say, there is quite a bit to consider
Should this feature exist at all? Does InfiniTime have sensitive data worthy of locking the device?
What are the semantics of locking? Which features should be disabled? Does it justify the extra complexity and code churn to have a locked state? Same goes for having a separate path for polling the heart rate sensor.
Is sensible locking actually possible with the heart rate sensor? - this one I can at least offer an opinion on: I'd say maybe. The sampling rate might have to be high enough to drain the battery noticeably. Also, tuning for different skin colours will make testing more challenging. Otherwise it's probably possible to set the HRS on a long delay and trigger some event if the value falls below a threshold for some number of readings.

**Comment by liamcharger** [comment]
I don't think it'd be very practical to use the HR sensor to achieve this. 

There are two reasons I would want this to be implemented: it would allow the watch to stop measuring heart rate when the watch isn't on the user's wrist (which would defeat the purpose if we use the HR sensor to check) and we'd be able to allow the companion app (InfiniLink) to check if it should send info notifications (low battery, goal achievement, etc.) or to just send them to host device.

Using the accelerometer to measure if it's being worn or not could be an idea (??), but it wouldn't be ideal as if the watch was moved around (picked up, etc.) it would register as being worn again, and if the user was asleep, etc., it would be registered as not on the wrist.

**Comment by jlariviere1** [comment]
Theoretically, if the watchface showed the detection, it could help the user wear the watch correctly.  I am less concerned with locking the watch, as I am with the watch knowing it is worn.  I've checked Gadgetbridge on mine, and several times, for the majority of the day it shows as not being active while I was wearing the watch.
